modules/currency-l0/src/main/scala/org/tessellation/currency/l0/snapshot/services/StateChannelSnapshotService.scala

```diff
@@ -33,37 +34,36 @@ trait StateChannelSnapshotService[F[_]] {
 object StateChannelSnapshotService {
   def make[F[_]: Async: KryoSerializer: SecurityProvider](
     keyPair: KeyPair,
-    lastSignedBinaryHashStorage: LastSignedBinaryHashStorage[F],
+    lastBinaryHashStorage: LastBinaryHashStorage[F],
     stateChannelSnapshotClient: StateChannelSnapshotClient[F],
     globalL0ClusterStorage: L0ClusterStorage[F],
     snapshotStorage: SnapshotStorage[F, CurrencyIncrementalSnapshot, CurrencySnapshotInfo]
   ): StateChannelSnapshotService[F] =
     new StateChannelSnapshotService[F] {
       private val logger = Slf4jLogger.getLogger
 
-      def createGenesisBinary(snapshot: Signed[CurrencySnapshot]): F[Signed[StateChannelSnapshotBinary]] = for {
-        lastSnapshotBinaryHash <- lastSignedBinaryHashStorage.get
-        bytes = JsonBinarySerializer.serialize(snapshot)
-        binary <- StateChannelSnapshotBinary(lastSnapshotBinaryHash, bytes, SnapshotFee.MinValue).sign(keyPair)
-      } yield binary
+      def createGenesisBinary(snapshot: Signed[CurrencySnapshot]): F[Signed[StateChannelSnapshotBinary]] = {
+        val bytes = JsonBinarySerializer.serialize(snapshot)
+        StateChannelSnapshotBinary(Hash.empty, bytes, SnapshotFee.MinValue).sign(keyPair)
+      }
 
       def createBinary(snapshot: Signed[CurrencySnapshotArtifact]): F[Signed[StateChannelSnapshotBinary]] = for {
-        lastSnapshotBinaryHash <- lastSignedBinaryHashStorage.get
+        lastSnapshotBinaryHash <- lastBinaryHashStorage.get
```

Comment:
if it's a hash of a binary and not the signed binary, won't it break the reference chain on Global L0?

Reply:
No, it was changed there too.